python/device/device.py

```diff
@@ -108,39 +115,100 @@ def __init__(self):
+            "processBytesInAllHeaps",
+            "All virtual memory used by the process",
+            "bytes",
+        )
+        self.reporter.add_integer_measurement(
+            "process_private_bytes",
+            "processPrivateBytes",
+            "Amount of non-shared physical memory used by the process",
+            "bytes",
+        )
+        self.reporter.add_integer_measurement(
+            "process_working_set_private",
+            "processWorkingSetPrivate",
+            "Amount of non-shared physical memory used by the process",
+            "bytes",
+        )
+        self.reporter.add_integer_measurement(
+            "total_messages_sent",
+            "totalMessagesSent",
+            "count of messages sent and ack'd by the transport",
+            "message(s)",
+        )
+        self.reporter.add_integer_measurement(
+            "total_messages_acknowledged",
+            "totalMessagesAcknowledged",
+            "count of messages sent to iothub with receipt verified via service sdk",
+            "message(s)",
+        )
+        self.reporter.add_integer_measurement(
+            "message_backlog",
+            "messageBacklog",
+            "count of messages waiting to be sent",
+            "message(s)",
+        )
+        self.reporter.add_integer_measurement(
+            "messages_in_flight",
+            "messagesInFlight",
+            "count of messages sent to iothub but not ack'd by the transport",
+            "message(s)",
+        )
+        self.reporter.add_integer_measurement(
+            "messages_unacknowledged",
+            "messagesUnacknowledged",
+            "count of messages sent to iothub and acked by the transport, but receipt not verified via service sdk",
+            "message(s)",
+        )
 
     def get_longhaul_metrics(self):
         self.metrics.run_time = (
             datetime.datetime.now(datetime.timezone.utc) - self.metrics.run_start_utc
         )
 
+        sent = self.metrics.d2c_sent.get_count()
+        acknowledged = self.metrics.d2c_acknowledged.get_count()
+
         props = {
             "runStartUtc": self.metrics.run_start_utc.isoformat(),
             "runEndUtc": self.metrics.run_end_utc.isoformat() if self.metrics.run_end_utc else None,
             "runTime": str(self.metrics.run_time),
             "runState": str(self.metrics.run_state),
             "exitReason": self.metrics.exit_reason,
-            "pingbacks": {
-                "requestsSent": self.metrics.pingback_requests_sent.get_count(),
-                "responsesReceived": self.metrics.pingback_responses_received.get_count(),
-            },
-            "d2c": {
-                "queued": self.telemetry_queue.qsize(),
-                "inFlight": self.metrics.d2c_in_flight.get_count(),
-                "sent": self.metrics.d2c_sent.get_count(),
-                "acknowledged": self.metrics.d2c_acknowledged.get_count(),
-            },
+            "totalMessagesSent": sent,
```

Comment:
I'm sending these values as telemetry only, not as props.

Reply:
Err, I mean to AppInsights. I could see sending them as telemetry though. They don't seem like values worth of twin props to me. What do you think?

Reply:
I'm putting them into twin props because it gives me an easy way to check status without doing an app insights query.  We can remove these later.

---
In reply to: [505743261](https://github.com/Azure/iot-sdk-longhaul/pull/15#discussion_r505743261) [](ancestors = 505743261)